plugins/ocm/README.md

```diff
@@ -1,10 +1,246 @@
+       env: PluginEnvironment,
+     ): Promise<Router> {
+       const builder = await CatalogBuilder.create(env);
+       ...
+   +   const ocm = ManagedClusterProvider.fromConfig(env.config, {
+   +     logger: env.logger,
+   +   });
+   +   builder.addEntityProvider(ocm);
+       ...
+       const { processingEngine, router } = await builder.build();
+       await processingEngine.start();
+       ...
+   +   await env.scheduler.scheduleTask({
+   +     id: 'run_ocm_refresh',
+   +     fn: async () => {
+   +       await ocm.run();
+   +     },
+   +     frequency: { minutes: 30 },
+   +     timeout: { minutes: 10 },
+   +   });
+       return router;
+     }
+   ```
+
+### Setup `@janus-idp/backstage-plugin-ocm`
+
+Install the frontend plugin:
+
+```sh
+yarn workspace app add @janus-idp/backstage-plugin-ocm
+```
+
+Then you can just pick whichever components you want to use.
+
+#### OcmPage
+
+A standalone page/dashboard showing all clusters as tiles.
+
+Add to `packages/app/src/App.tsx`:
+
+```diff
++ import { OcmPage } from '@janus-idp/backstage-plugin-ocm';
+
+  const routes = (
+    <FlatRoutes>
+      ...
++     <Route path="/ocm" element={<OcmPage logo={<Logo />} />} />
+      ...
+    </FlatRoutes>
+  );
+```
+
+And update navigation in `packages/app/src/components/Root/Root.tsx`:
+
+```diff
++ import StorageIcon from '@material-ui/icons/Storage';
+  ...
+  export const Root = ({ children }: PropsWithChildren<{}>) => (
+    <SidebarPage>
+      <Sidebar>
+        <SidebarGroup label="Menu" icon={<MenuIcon />}>
+          ...
++         <SidebarItem icon={StorageIcon} to="ocm" text="Clusters" />
+          ...
+       </SidebarGroup>
+       ...
+      </Sidebar>
+      {children}
+    </SidebarPage>
+  );
+```
+
+#### ClusterContextProvider
+
+React context provided for OCM data related to current entity. This component is required to be used in order to display any data on following React components.
+
+Place this context provider into your `Resource` entity rendeder (usually in `packages/app/src/components/catalog/EntityPage.tsx` or in a component imported by in here):
+
+```diff
++ import { ClusterContextProvider } from '@janus-idp/backstage-plugin-ocm';
+  ...
+ const isType = (types: string | string[]) => (entity: Entity) => {
+  if (!entity?.spec?.type) {
+    return false;
+  }
+  return typeof types === 'string'
+    ? entity?.spec?.type === types
+    : types.includes(entity.spec.type as string);
+};
+
++ const resourcePage = {
+    ...
++           <EntitySwitch.Case if={e => e?.spec?.type === "kubernetes-cluster"}>
++             <ClusterContextProvider>
+                ... // OCM plugin components goes here
++             </ClusterContextProvider>
++           </EntitySwitch.Case>
+    ...
++ }
+  ...
+  export const entityPage = (
+    <EntitySwitch>
+      ...
++     <EntitySwitch.Case if={isKind('resource')} children={resourcePage} />
+      ...
+    </EntitySwitch>
+  );
+
+
+```
+
+#### ClusterStatusCard
```

Comment:
should we add example on how to use these cards

Reply:
I've added that. since it's very layout dependent, it's not very helpful though.